### PR TITLE
Forward ProcessException to error handlers

### PR DESCRIPTION
**Pull request by blasten**
## Description

Forwards Permission denied to the error handlers.

## Tests

I added the following tests: Unit test

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -338,35 +338,40 @@ Future<void> buildGradleApp({
...
-            break;
-          }
-        }
-        // Pipe stdout/stderr from Gradle.
-        return line;
-      },
+      mapFunction: consumeLog,
     );
+  } on ProcessException catch(exception) {
+    consumeLog(exception.toString());
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/gradle.dart]: Do you need to rethrow if `consumeLog` doesn't handle the exception?

Comment (blasten) [reply packages/flutter_tools/lib/src/android/gradle.dart]: True. Thanks

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 2]
```diff
@@ -334,39 +334,48 @@ Future<void> buildGradleApp({
   }
   command.add(assembleTask);
 
-  final Stopwatch sw = Stopwatch()..start();
-  int exitCode = 1;
   GradleHandledError detectedGradleError;
   String detectedGradleErrorLine;
+  final StringConverter consumeLog = (String line) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/gradle.dart]: Nit: prefer a (mostly) consistent style of function declaration.

```suggestion
  String consumeLog(String line) {
```

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**